src/main/java/prefs/Preferences.java

```diff
@@ -10,160 +16,296 @@
     }
 
-    // Last login credentials. While the main UI is running (i.e. logged in successfully), last login
-    // credentials are guaranteed to be the current user's credentials thanks to setLastLoginCredentials
-    // being called immediately after a successful login in LoginDialog.
+    /**
+     * Retrieves the last login password
+     */
     public String getLastLoginPassword() {
-        return global.getLastLoginPassword();
+        return sessionConfig.getLastLoginPassword();
     }
 
+    /**
+     * Retrieves the last login username
+     */
     public String getLastLoginUsername() {
-        return global.getLastLoginUsername();
+        return sessionConfig.getLastLoginUsername();
     }
 
+    /**
+     * Sets the last login credentials
+     * @param username The username used during the last login
+     * @param password The password used during the last login
+     */
     public void setLastLoginCredentials(String username, String password) {
-        global.setLastLoginCredentials(username, password);
-    }
-
-    public List<String> getLastOpenFilters() {
-        return global.getLastOpenFilters();
-    }
-
-    public List<String> getPanelNames() {
-        return global.getPanelNames();
+        sessionConfig.setLastLoginCredentials(username, password);
+        save();
     }
 
+    /**
+     * Retrieves a list of panel infos
+     */
     public List<PanelInfo> getPanelInfo() {
-        return global.getPanelInfo();
-    }
-
-    public void setPanelInfo(List<PanelInfo> panelInfo) {
-        global.setPanelInfo(panelInfo);
+        return sessionConfig.getPanelInfo();
     }
 
     /**
-     * Interface to configuration files
+     * Sets the panel infos
+     * @param panelInfo The list of panel infos to set to
      */
+    public void setPanelInfo(List<PanelInfo> panelInfo) {
+        sessionConfig.setPanelInfo(panelInfo);
+        save();
+    }
 
     /**
-     * Boards
+     * Adds a board with the board name and its panels information
      */
-
     public void addBoard(String name, List<PanelInfo> panels) {
         assert name != null && panels != null;
-        global.addBoard(name, panels);
+        sessionConfig.addBoard(name, panels);
+        save();
     }
 
+    /**
+     * Retrieves all the boards
+     */
     public Map<String, List<PanelInfo>> getAllBoards() {
-        return global.getAllBoards();
+        return sessionConfig.getAllBoards();
     }
 
+    /**
+     * Retrieves a list of all the board names
+     */
     public List<String> getAllBoardNames() {
         return new ArrayList<>(getAllBoards().keySet());
     }
 
+    /**
+     * Removes the board specified
+     * @param name The name of the board to remove
+     */
     public void removeBoard(String name) {
-        global.removeBoard(name);
+        sessionConfig.removeBoard(name);
+        save();
     }
 
+    /**
+     * Sets the last open board
+     * @param board The name of the last open board.
```

Comment:
Instead of having the `@param`, would be simpler to refactor the variable name of the parameter to `boardName`, which would make this header comment unnecessary. Same with above.